### ISSUE TITLE
Reproducing results from WEASEL paper

### ISSUE THREAD

**Issue by bkj** [johannfaouzi/pyts#24]
Hi --

Do you have an example of using WEASEL to get results close to those reported in [the paper](https://arxiv.org/pdf/1701.07681.pdf)?  I'm playing around w/ using WEASEL as a featurizer on some of the UCR datasets, but my results aren't nearly as good as in the paper -- I'm guessing I'm not using the right hyperparameters.  Any ideas?

Thanks!

**Comment by johannfaouzi** [comment]
Hi,

I created another repository where I compare the accuracies with the implementations in **pyts** and the accuracies reported in the original papers: [pyts-repro](https://github.com/johannfaouzi/pyts-repro).

More specifically about WEASEL, here is the [corresponding notebook](https://github.com/johannfaouzi/pyts-repro/blob/master/0.8.0/WEASEL.ipynb).

Please note that the default parameters in **pyts** are different from the ones used in the original implementation, especially regarding the window sizes:
* pyts: [[0.1, 0.3, 0.5, 0.7, 0.9]](https://github.com/johannfaouzi/pyts/blob/aa3e33f6311cfdf79ad6357e8ba18c12b4f571b4/pyts/transformation/weasel.py#L83-L86) (expressed as a percentage of the length of the time series),
* original implementation : [from 2 to 350](https://github.com/patrickzib/SFA/blob/0aa1fbdd4ae6eb5103de85d96b56574a34a05513/src/main/java/sfa/classification/WEASELClassifier.java#L40-L41).

Another important hyper-parameter is the word size, which is chosen with cross-validation [from 4 to 6 in the original implementation](https://github.com/patrickzib/SFA/blob/0aa1fbdd4ae6eb5103de85d96b56574a34a05513/src/main/java/sfa/classification/WEASELClassifier.java#L26-L27).

I hope that it helps a bit.

Johann

**Comment by bkj** [comment]
Thanks! 

I assume the values for word and window size are tuned via cross validation in that notebook?

**Comment by johannfaouzi** [comment]
As mentioned in the [introduction in the README](https://github.com/johannfaouzi/pyts-repro#introduction), I did not perform an unbiased cross-validation to find the best values for the hyper-parameters (because I used the test set to find them), thus the numbers reported in the **pyts** column are the minimum of the results reported in the paper and the results obtained with a biased cross-validation.

What I strive to provide with this package is **reusable code**, with a lot of time spent on writing proper and readable code, and precise documentation. For this reason, I focus less on **strict reproducibility** of the original papers (several algorithms have a built-in cross-validation procedure, and I prefer not implementing it and letting the user do it on their own). If the performance with **pyts** is comparable with the performance reported in the paper, I think that it is enough (at least for now).
